crates/router/src/core/co_badged_cards_info.rs

```
@@ -0,0 +1,153 @@
+            .attach_printable("The filtered co-badged card info list is empty")?;
+        Ok(first_element.regulated_name.clone())
+    }
+
+    pub fn get_co_badged_cards_info_response(
+        &self,
+    ) -> CustomResult<CoBadgedCardInfoResponse, errors::ApiErrorResponse> {
+        Ok(CoBadgedCardInfoResponse {
+            card_networks: self.extract_networks(),
+            regulated: self.is_regulated()?,
+            regulated_name: self.get_regulated_name()?,
+        })
+    }
+}
+
+pub async fn get_co_badged_cards_info(
+    state: routes::SessionState,
+    key_store: domain::MerchantKeyStore,
+    card_number: cards::CardNumber,
+) -> CustomResult<Option<CoBadgedCardInfoResponse>, errors::ApiErrorResponse> {
+    let db = state.store.as_ref();
+    let key_manager_state = &(&state).into();
+
+    // The card number is padded to 16 digits as the co-badged card bin info is stored as a 16 digit number
+    let card_number_str = CoBadgedCardInfoList::pad_card_number_to_16_digit(card_number);
+
+    let parsed_number: i64 = card_number_str
+        .parse::<i64>()
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable(
+            "Failed to convert card number to integer in co-badged cards info flow",
+        )?;
+
+    let co_badged_card_infos_record = db
+        .find_co_badged_cards_info_by_card_bin(key_manager_state, &key_store, parsed_number)
+        .await;
+
+    let filtered_co_badged_card_info_list_optional = match co_badged_card_infos_record {
+        Err(error) => {
+            if error.current_context().is_db_not_found() {
+                logger::debug!("co-badged card info record not found");
+                Ok(None)
+            } else {
+                Err(error)
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("error while fetching co-badged card info record")
```

Should we throw an error here? Are we catching it in the debit routing core? Cuz we can probably just skip debit routing in case of an error without impeding the payment. Check for other errors in this function as well.

---

This error would occur if the record is present in the database but could not be retrieved. I want to return the error from this function and handle the decision to skip this part in the function caller.